### PR TITLE
restore database from cloud

### PR DESCRIPTION
**Pull request by xiaojingchen**
this PR add restore job, the feature can restore data to a TiDB cluster, the premise is that the data is saved in the cloud (recommended gcs)

### CODE REVIEW

**Comment by gregwebs** [comment]
What's the plan to make this scale?

1) Download one file first
2) Start loading the first file while downloading the second
3) Delete the first file once it is loaded.

File: charts/tidb-cluster/templates/scripts/_start_restore.sh.tpl [hunk 1]
```diff
@@ -0,0 +1,22 @@
+set -euo pipefail
+
+dirname=`date +%Y-%m-%dT%H%M%S`-${MY_POD_NAME}
+mkdir -p /data/${dirname}/
```
Comment (weekface) [inline charts/tidb-cluster/templates/scripts/_start_restore.sh.tpl]: difine `/data/${dirname}/` as a var.

File: charts/tidb-cluster/templates/scripts/_start_restore.sh.tpl [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+{{- if .Values.restore.gcp }}
+downloader \
+  --cloud=gcp \
+  --bucket={{ .Values.restore.gcp.bucket }} \
+  --srcDir={{ .Values.restore.gcp.srcDir }} \
+  --destDir=/data/${dirname}
+
+dataDir=/data/${dirname}/{{ .Values.restore.gcp.srcDir }}
+./loader \
```
Comment (weekface) [inline charts/tidb-cluster/templates/scripts/_start_restore.sh.tpl]:
```suggestion
loader \
```
?

**Comment by tennix** [comment]
@gregwebs I think it's a long-term plan to make `loader` support cloud natively.
/cc @GregoryIan

File: charts/tidb-cluster/templates/scripts/_start_fullbackup.sh.tpl [hunk 3]
```diff
@@ -5,7 +5,7 @@ host=`echo {{ .Values.clusterName }}_TIDB_SERVICE_HOST | tr '[a-z]' '[A-Z]'`
 mkdir -p /data/${dirname}/
 cp /savepoint-dir/savepoint /data/${dirname}/
 
-mydumper \
+./mydumper \
```
[the remark this answers is not in the document]
Comment (xiaojingchen) [reply charts/tidb-cluster/templates/scripts/_start_fullbackup.sh.tpl]: new version  image: `pingcap/tidb-cloud-backup:latest` will base `pingcap/tidb-enterprise-tools:latest` , the `mydumper` and `loader`'s path change.
look at:
https://github.com/tennix/tidb-cloud-backup/pull/1

File: charts/tidb-cluster/templates/scripts/_start_restore.sh.tpl (hunk 2 above)
Comment (xiaojingchen) [reply charts/tidb-cluster/templates/scripts/_start_restore.sh.tpl]: ditto

**Comment by xiaojingchen** [comment]
@gregwebs   the plan is:
1. download all the data files from gcs to pv first
2. the loader restore the datas to tidb

if any step failed then the job is failed and quit

File: charts/tidb-cluster/templates/scripts/_start_fullbackup.sh.tpl (hunk 3 above)
[the remark this answers is not in the document]
Comment (tennix) [reply charts/tidb-cluster/templates/scripts/_start_fullbackup.sh.tpl]: If so, please use an absolute path.

File: charts/tidb-cluster/templates/restore-job.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      {{- if .Values.restore.gcp }}
+        - name: GOOGLE_APPLICATION_CREDENTIALS
+          value: /gcp/credentials.json
+      {{- end }}
+        - name: TIDB_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.clusterName }}-restore-secret
+              key: user-password
+        restartPolicy: OnFailure
```
Comment (gregwebs) [inline charts/tidb-cluster/templates/restore-job.yaml]: This is here a second time.

File: charts/tidb-cluster/templates/scripts/_start_restore.sh.tpl [hunk 5]
```diff
@@ -0,0 +1,23 @@
+set -euo pipefail
+
+dirname=`date +%Y-%m-%dT%H%M%S`-${MY_POD_NAME}
+dataDir=/data/${dirname}
+mkdir -p ${dataDir}/
+host=`echo {{ .Values.clusterName }}_TIDB_SERVICE_HOST | tr '[a-z]' '[A-Z]'`
+
+{{- if .Values.restore.gcp }}
+downloader \
```
Comment (gregwebs) [inline charts/tidb-cluster/templates/scripts/_start_restore.sh.tpl]: This fails for me. `/bin/sh: line 7: downloader: command not found`

Comment (tennix) [reply charts/tidb-cluster/templates/scripts/_start_restore.sh.tpl]: The latest image is just pushed to DockerHub.

**Pull request merged**